File: Nov2020/20Nov/MakeTheStringGreat.java
/*
Problem Name: 1544. Make The String Great
Problem Link: https://leetcode.com/problems/make-the-string-great/
*/
class Solution {
    public String makeGood(String s) {
        Stack<Character> st = new Stack<Character>();
        int diff = 'a' - 'A';
        for(char c: s.toCharArray()){
            if(st.isEmpty())
                st.push(c);
            else if(Math.abs(st.peek() - c) == diff)
                st.pop();
            else
                st.push(c);
        }
        s = "";
        while(!st.isEmpty())
            s = st.pop() + s;
        return s;
    }
}

File: Nov2020/20Nov/MakeTheStringGreat.cpp
/*
Problem name: 1544. Make The String Great
Problem link: https://leetcode.com/problems/make-the-string-great/
*/

class Solution {
public:
    string makeGood(string s) {
        int n=s.length();
        for(int i=0;i<n-1;++i){
            if(i>=0 && tolower(s[i])==tolower(s[i+1])){
                if((islower(s[i]) && isupper(s[i+1])) || (isupper(s[i]) && islower(s[i+1]))) {
                    s.erase(i,2);
                    i=i-2;
                }
            }
        }
        return s;
    }
};
